src/chat/UnreadNotice.js

```diff
@@ -54,56 +60,97 @@ export default class UnreadNotice extends React.Component {
-    Animated.timing(this.state.translateAnimation, hideAnimationConfig).start();
-  };
+    Animated.timing(this.state.translateAnimation,
+      Object.assign({}, hideAnimationConfig, { toValue: unreadCount === 0 ? 0 : 0.4 })
+    ).start(() => {
+      this.setState({
+        noticeState: this.state.translateAnimation._value === 0 ?
+          noticeStates.hidden :
+          noticeStates.peeking
+      });
+    });
+  }, DEBOUNCE_DURATION);
+
+  hidePeekingNotice = debounce(() => {
+    this.state.translateAnimation.setValue(0.5);
+    Animated.timing(this.state.translateAnimation, hideAnimationConfig).start(() => {
+      this.setState({
+        noticeState: noticeStates.hidden
+      });
+    });
+  }, DEBOUNCE_DURATION);
 
-  show = () => {
+  show = debounce(() => {
     this.state.translateAnimation.setValue(0);
-    Animated.timing(this.state.translateAnimation, showAnimationConfig).start();
-  };
+    Animated.timing(this.state.translateAnimation, showAnimationConfig).start(() => {
+      this.setState({
+        noticeState: noticeStates.visible
+      });
+    });
+
+    // Auto-hide notice HIDE_DELAY milliseconds after it's shown
+    clearTimeout(this.hideTimeout);
+    this.hideTimeout = setTimeout(this.hide, HIDE_DELAY);
+  }, DEBOUNCE_DURATION);
 
   dynamicContainerStyles = () => {
-    const { position, shouldOffsetForInput } = this.props;
-    const translationMultiplier = position === POSITIONS.top ? -1 : 1;
+    const { shouldOffsetForInput } = this.props;
+    // In narrows where ComposeBox is present translate beyond ComposeBox without blocking it
+    const translateTo = shouldOffsetForInput ? -40 : 0;
+    const translateFrom = shouldOffsetForInput ? 0 : 50;
 
     // In narrows where ComposeBox is present translate beyond ComposeBox to avoid blocking it
     const translateTo = shouldOffsetForInput && position === POSITIONS.bottom ? -40 : 0;
     const translateFrom = shouldOffsetForInput && position === POSITIONS.bottom ? 0 : 50;
 
     return {
       ...StyleSheet.flatten(styles.unreadContainer),
-      bottom: position === POSITIONS.bottom ? 0 : null,
-      top: position === POSITIONS.top ? 0 : null,
+      bottom: 0,
       transform: [
         {
           translateY: this.state.translateAnimation.interpolate({
             inputRange: [0, 1],
-            outputRange: [translationMultiplier * translateFrom, translateTo]
+            outputRange: [translateFrom, translateTo]
           })
         }
       ]
     };
   };
 
   render() {
-    const { count } = this.props;
-
+    const { unreadCount } = this.props;
     return (
       <Animated.View style={this.dynamicContainerStyles()}>
-        <IconDownArrow style={[styles.icon, styles.downArrowIcon]} />
+        <IconDownArrow style={styles.icon} />
         <Text style={styles.unreadText}>
-          {count === 0 ?
-            'No' : count < 100 ?
-              count : '99+'} unread {count === 1 ? 'message' : 'messages'}
+          {unreadCount === 0 ?
```

Comment:
We should consider how to translate that. I might do it in another commit after we merge this one.